http/design/response.go

```diff
@@ -210,3 +214,31 @@ func (r *HTTPResponseExpr) Dup() *HTTPResponseExpr {
 		headers:     design.DupAtt(r.headers),
 	}
 }
+
+// bodyAllowedForStatus reports whether a given response status code
+// permits a body. See RFC 2616, section 4.4.
+// See https://golang.org/src/net/http/transfer.go
+func bodyAllowedForStatus(status int) bool {
+	switch {
+	case status >= 100 && status <= 199:
+		return false
+	case status == 204:
+		return false
+	case status == 304:
+		return false
+	}
+	return true
+}
+
+// bodyExists returns true if a response body is defined in the
+// response expression via Body() or Result() in the method expression.
```

Comment:
It seems the function returns false if the response has a body but is defined at the service or API level. If this is OK it should probably be documented in the comment?

Reply:
HTTP expression at the API or Service level can only contain error response as per the documentation. And all HTTP error codes can contain a response body. So I think we can have this function just check the method expression for now.